app/controllers/staff/base_controller.rb

```diff
@@ -1,8 +1,17 @@
 class Staff::BaseController < ApplicationController
+  rescue_from Pundit::NotAuthorizedError, with: :pundit_not_authorized
```

Comment:
We recently created an `Authorisation` module to handle the case when the authenticated user doesn't exist locally. It catches and renders (rather than redirects) to a custom error page that explains the problem and sets some expectations about what to do if this is unexpected.

I think there is an opportunity to merge these 2 error handling flows together into that concern. Would it be possible to render an error page here too instead of abruptly redirecting the user all the way back to the dashboard. From the render they can at least go 'back' in the browser to get back to where they went wrong to prevent any lost progress or flow.

Reply:
:+1: